src/store/Controller.cc

```diff
@@ -885,33 +870,37 @@ Store::Controller::anchorToCache(StoreEntry &entry, bool &inSync)
     Transients::EntryStatus entryStatus;
     transients->status(entry, entryStatus);
 
+    inSync = false;
     bool found = false;
     if (sharedMemStore)
         found = sharedMemStore->anchorToCache(entry, inSync);
     if (!found && swapDir)
         found = swapDir->anchorToCache(entry, inSync);
 
+    if (inSync) {
+        debugs(20, 7, "anchored " << entry);
+        assert(found);
+        entry.setCollapsingRequirement(false);
+        return true;
+    }
+
     if (found) {
-        if (inSync)
-            debugs(20, 7, "anchored " << entry);
-        else
-            debugs(20, 5, "failed to anchor " << entry);
+        debugs(20, 5, "failed to anchor " << entry);
         return true;
     }
 
     if (entryStatus.waitingToBeFreed) {
         debugs(20, 5, "failed on marked unattached " << entry);
-        inSync = false;
         return true;
```

Comment:
The pair of `found` and `inSync` flags is difficult to interpret correctly because the two are not independent. This PR does not change that -- it only improves symmetry in flags handling across anchorToCache() and syncCollapsed() methods. We are working on a followup PR that fixes another bug and eliminates the `inSync` flag from this method.